pkg/cmd/pr/comment/comment.go

```diff
@@ -0,0 +1,78 @@
+package comment
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/context"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/pr/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdComment(f *cmdutil.Factory, runF func(*shared.CommentableOptions) error) *cobra.Command {
+	opts := &shared.CommentableOptions{
+		IO:                    f.IOStreams,
+		HttpClient:            f.HttpClient,
+		EditSurvey:            shared.CommentableEditSurvey(f.Config, f.IOStreams),
+		InteractiveEditSurvey: shared.CommentableInteractiveEditSurvey(f.Config, f.IOStreams),
+		ConfirmSubmitSurvey:   shared.CommentableConfirmSubmitSurvey,
+		OpenInBrowser:         utils.OpenInBrowser,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "comment [<number> | <url> | <branch>]",
+		Short: "Create a new pr comment",
+		Example: heredoc.Doc(`
+			$ gh pr comment 22 --body "This looks great, lets get it deployed."
+		`),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
I've never figured out a good use-case for Cobra's hooks. I like that you're exploring this here. But to help me understand: is this PreRunE block here because the same logic wouldn't work (or wouldn't be as testable) if it were just moved to the RunE block?

Reply:
I used the `PreRunE` block here mostly for organization and readability of the code. The same code could be moved into the `RunE` block and everything would function the exact same. I felt that in the `Commentable` file we needed a clean way to differentiate between the configuring of the command and the running of the command, this distinction is easier to follow when initializing the `cobra.Command` but got lost when extracting the common code to `Commentable`. Overall, this direction felt cleaner and created an easy mapping between `Commentable` and the PR comment, issue comment commands.